transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java

```diff
@@ -338,6 +338,19 @@ private void addPollOut() {
 
     abstract class AbstractUringUnsafe extends AbstractUnsafe {
         private IOUringRecvByteAllocatorHandle allocHandle;
+        private ChannelPromise delayedClose;
+
+        @Override
+        public void close(ChannelPromise promise) {
+            if ((ioState & WRITE_SCHEDULED) == 0) {
```

Comment:
not sure what you suggest in terms of moving... can you clarify ?

yes i guess we need all ops to make it rubust

Reply:
> not sure what you suggest in terms of moving... can you clarify ?

Oops sorry I meant the check where the deferred close is done, i.e. this logic: 
https://github.com/1Jo1/netty/pull/38/files#diff-f6c6f00f86dd1739b0d13ebca4e19567R579-R584

Since the end of the handle() method is a single place where processing of CQEs is completed. So the check there can be `if (delayedClose != null && all _SCHEDULED bits are 0) { ...`.

And have each of the `AbstractIOUringChannel#doX()` methods only submit if not in closed state.